rali/rali/source/image_read_and_decode.cpp

```diff
@@ -132,78 +133,101 @@ ImageReadAndDecode::load(unsigned char* buff,
     const Decoder::ColorFormat decoder_color_format = std::get<0>(ret);
     const unsigned output_planes = std::get<1>(ret);
     const bool keep_original = decoder_keep_original;
+    const size_t image_size = max_decoded_width * max_decoded_height * output_planes * sizeof(unsigned char);
 
     // Decode with the height and size equal to a single image  
     // File read is done serially since I/O parallelization does not work very well.
     _file_load_time.start();// Debug timing
-
-    while ((file_counter != _batch_size) && _reader->count() > 0)
-    {
-        size_t fsize = _reader->open();
-        if (fsize == 0) {
-            WRN("Opened file " + _reader->id() + " of size 0");
-            continue;
+    if (_decoder_config._type == DecoderType::SKIP_DECODE) {
+        while ((file_counter != _batch_size) && _reader->count() > 0)
+        {
+            auto read_ptr = buff + image_size * file_counter;
+            size_t fsize = _reader->open();
+            if (fsize == 0) {
+                WRN("Opened file " + _reader->id() + " of size 0");
+                continue;
+            }
+
+            _actual_read_size[file_counter] = _reader->read(read_ptr, fsize);
```

Comment:
Codacy found an issue: [Check buffer boundaries if used in a loop including recursive loops (CWE-120, CWE-20).  ](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/pullRequest?prid=6366586)